### PR TITLE
[Bug] pa_mqa_logits: guard all OutLogits stores

### PR DESCRIPTION
**Pull request by maeehart**
## Summary

- Add `offset < max_model_len` to the eight `OutLogits_buffer` stores that remained lower-bound-only after #2866 landed through #3005.
- Cover the two non-preshuffle stores, three fixed-context preshuffle stores, and three varctx preshuffle stores.
- Add structural coverage for every OutLogits store and ROCm guard-region regression cases for fixed-context and varctx kernels.

This replaces #2936 with a minimal diff against current main.

## Test plan

- [x] `python -m pytest -q op_tests/test_pa_mqa_logits_store_bounds.py` (`3 passed, 2 skipped` locally because ROCm is unavailable)
- [x] `python -m black --check aiter/ops/triton/gluon/pa_mqa_logits.py op_tests/test_pa_mqa_logits_store_bounds.py`
- [x] `python -m ruff check aiter/ops/triton/gluon/pa_mqa_logits.py op_tests/test_pa_mqa_logits_store_bounds.py`
- [ ] Run the fixed-context and varctx guard-region cases on ROCm CI

### CODE REVIEW

File: op_tests/test_pa_mqa_logits_store_bounds.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    from aiter import dtypes
+    from aiter.ops.triton.pa_mqa_logits import (
+        deepgemm_fp8_paged_mqa_logits,
+    )
+
+    context_len = 2300
+    block_size = 64
+    chunk_k = 256
+    guard_size = block_size
+    hidden_dim = 128
```
[the remark this answers is not in the document]
Comment (maeehart) [reply op_tests/test_pa_mqa_logits_store_bounds.py:106]: Fixed in [`51c2c7e56`](https://github.com/ROCm/aiter/commit/51c2c7e567b861b2251024cbddd5f1eac9a97f70). The guard now uses `ChunkK=256`, which covers the tested maximum overshoot of 190 elements, so a mask regression is caught by the sentinel assertion instead of writing beyond the allocation. The same commit removes the unnecessary shebang and applies the repository import ordering.

Focused validation passes with 3 host tests. The 2 ROCm cases are left to the PR GPU workflow.
